packages/frontend/src/components/ChartConfigPanel/ChartConfigTabs.tsx

```diff
@@ -15,12 +15,82 @@ import {
 } from '@lightdash/common';
 import React, { FC, useCallback, useMemo, useState } from 'react';
 import { useOrganisation } from '../../hooks/organisation/useOrganisation';
+import { useTracking } from '../../providers/TrackingProvider';
+import { EventName } from '../../types/Events';
 import { useVisualizationContext } from '../LightdashVisualization/VisualizationProvider';
-import { InputWrapper, Wrapper } from './ChartConfigPanel.styles';
+import {
+    InputWrapper,
+    MinMaxContainer,
+    MinMaxInput,
+    MinMaxWrapper,
+    Wrapper,
+} from './ChartConfigPanel.styles';
 import FieldLayoutOptions from './FieldLayoutOptions';
 import BasicSeriesConfiguration from './Series/BasicSeriesConfiguration';
 import GroupedSeriesConfiguration from './Series/GroupedSeriesConfiguration';
 
+interface MinMaxProps {
+    minVal: (index: number, value: string | number | undefined) => void;
+    maxVal: (index: number, value: string | number | undefined) => void;
+    index: number;
+    axis: string;
+}
+
+const AxisMinMax: FC<MinMaxProps> = ({ minVal, maxVal, index, axis }) => {
+    const [isAuto, setIsAuto] = useState<boolean>(true);
+    const {
+        cartesianConfig: { dirtyEchartsConfig },
+    } = useVisualizationContext();
+    const { track } = useTracking();
+
+    const minDefaultValue =
+        (axis === 'y'
+            ? dirtyEchartsConfig?.yAxis?.[index]?.min?.toString()
+            : dirtyEchartsConfig?.xAxis?.[index]?.min?.toString()) || undefined;
+    const maxDefaultValue =
+        (axis === 'y'
+            ? dirtyEchartsConfig?.yAxis?.[index]?.max?.toString()
+            : dirtyEchartsConfig?.xAxis?.[index]?.max?.toString()) || undefined;
+
+    return (
+        <MinMaxContainer>
+            <Switch
+                name="auto-range"
+                checked={isAuto}
+                label={`Auto ${axis}-axis range`}
+                onChange={() => {
+                    setIsAuto((prev) => !prev);
+                    track({
+                        name: EventName.CUSTOM_AXIS_RANGE_TOGGLE_CLICKED,
+                        properties: {
+                            custom_axis_range: isAuto,
+                        },
+                    });
+                }}
+            />
+            {!isAuto && (
+                <MinMaxWrapper>
+                    <MinMaxInput label="Min">
+                        <InputGroup
+                            placeholder="Min"
+                            defaultValue={minDefaultValue}
+                            onBlur={(e) => minVal(index, e.currentTarget.value)}
```

Comment:
Isn't the value always string and eCharts supports a number as string "5" ? if so, should we only support min/max property as `string | undefined` ?